ci/scripts/utils/rocotostat.py

```diff
@@ -39,60 +41,137 @@ def input_args():
     return args
 
 
-def rocoto_statcount():
-    """
-    Run rocotostat and process its output.
+def rocotostat_summary(rocotostat):
     """
+    ROCOTO_SUMMARY Run rocotostat and process its output.
 
-    args = input_args()
+    ROCOTO_SUMMARY(ROCOTOSTAT) adds a default argument '--summary' to the rocotostat
+    command, runs it, and processes its output to return a dictionary with the total
+    number of cycles and the number of cycles marked as 'Done'.
 
-    try:
-        rocotostat = which("rocotostat")
-    except CommandNotFoundError:
-        logger.exception("rocotostat not found in PATH")
-        raise CommandNotFoundError("rocotostat not found in PATH")
+    Input:
+    ROCOTOSTAT - The rocotostat command.
 
-    rocotostat_all = which("rocotostat")
-    rocotostat.add_default_arg(['-w', os.path.abspath(args.w.name), '-d', os.path.abspath(args.d.name), '-s'])
-    rocotostat_all.add_default_arg(['-w', os.path.abspath(args.w.name), '-d', os.path.abspath(args.d.name), '-a'])
+    Output:
+    ROCOTO_STATUS - A dictionary with the total number of cycles and the number of cycles marked as 'Done'.
+    """
 
+    rocotostat.add_default_arg('--summary')
```

Comment:
Worried about side effects here. Python is "pass-by-assignment", which is mostly like pass-by-reference, so this is modifying the `Executable` object `rocotostat` is pointing to. If it is then used again by the calling routine, it will still have this argument. We probably want to make a copy here to modify locally. @aerorahul?